rwengine/src/ai/CharacterController.cpp

```diff
@@ -252,12 +252,12 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
                     character->getCurrentState().armour = 100.f;
                 } else if (vehicle->getVehicle()->vehiclename_ == "AMBULAN") {
                     // Receive up to 25 HPs from Ambulance
-                    if (character->getCurrentState().health < 100.f) {
-                        if (character->getCurrentState().health > 75.f) {
-                            character->getCurrentState().health = 100.f;
-                        } else {
-                            character->getCurrentState().health += 25.f;
-                        }
+                    float health = character->getCurrentState().health;
+                    // todo: Replace by correct fullHealth once implemented
+                    float fullHealth = 100.f;
+                    if (health < fullHealth) {
+                        character->getCurrentState().health =
+                            (health >= 80.f) ? fullHealth : health + 20.f;
```

Comment:
even with a 100% game [at which point you can have more than 100 HP from health pickups]?
You can always have more than 100% health from prostitutes.
Also good to know about the flag still being cleared :)